packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard.js

```diff
@@ -392,13 +388,25 @@ export const useCreditCard = props => {
     useEffect(() => {
         try {
             if (shouldSubmit) {
-                setStepNumber(1);
-                if (isBillingAddressSame) {
-                    setShippingAddressAsBillingAddress();
+                /**
+                 * Validate billing address fields and only process with
+                 * submit if there are no errors.
+                 */
+                validateBillingAddressForm();
```

Comment:
Is there no way to validate these fields on blur? Check out how [other forms](https://github.com/magento/pwa-studio/blob/2a56b839666426b8edb50964778e70a7488c0e4c/packages/venia-ui/lib/components/CreateAccount/createAccount.js#L77) do it. I'm not sure why we didn't use validation on the old mini cart.

Reply:
Discussed this on the call, commenting for notes.

We can not use the same techniques as in the previous forms, because of the way the form is submitted. In other forms, we submit using the `onSubmit` prop on the form so the validation is done by the form. In this case, we submit manually, form just serves as a context provider.  Hence we have to perform the form validation manually before submitting it.